scripts/cat/cats.py

```diff
@@ -819,79 +824,67 @@ def add_to_clan(self) -> list:
             child = Cat.all_cats[child_id]
             if (
-                child.outside
-                and not child.exiled
-                and not child.dead
-                and child.moons < 12
+                    child.outside
+                    and not child.exiled
+                    and not child.dead
+                    and child.moons < 12
             ):
                 child.add_to_clan()
                 History.add_beginning(child)
                 ids.append(child_id)
 
         return ids
 
-    def status_change(self, new_status, resort=False):
+    def status_change(self, new_status: StatusEnum | str, resort=False):
         """Changes the status of a cat. Additional functions are needed if you want to make a cat a leader or deputy.
-        new_status = The new status of a cat. Can be 'apprentice', 'medicine cat apprentice', 'warrior'
-                    'medicine cat', 'elder'.
+        new_status = The new status of a cat.
         resort = If sorting type is 'rank', and resort is True, it will resort the cat list. This should
                 only be true for non-timeskip status changes."""
+
+        # auto-converting string
+        if isinstance(new_status, str):
+            new_status = StatusEnum(new_status)
+
         old_status = self.status
         self.status = new_status
         self.name.status = new_status
 
+        if new_status.is_outside_clan():
+            return
+
         self.update_mentor()
         for app in self.apprentice.copy():
             fetched_cat = Cat.fetch_cat(app)
             if isinstance(fetched_cat, Cat):
                 fetched_cat.update_mentor()
 
         # If they have any apprentices, make sure they are still valid:
-        if old_status == "medicine cat":
+        if old_status.is_medcat():
             game.clan.remove_med_cat(self)
 
         # updates mentors
-        if self.status == "apprentice":
+        if new_status.is_app_any():
             pass
 
-        elif self.status == "medicine cat apprentice":
+        elif new_status.is_mediator_any():
             pass
 
-        elif self.status == "warrior":
-            if old_status == "leader":
-                if game.clan.leader:
-                    if game.clan.leader.ID == self.ID:
-                        game.clan.leader = None
-                        game.clan.leader_predecessors += 1
-
-                    # don't remove the check for game.clan, this is needed for tests
-            if game.clan and game.clan.deputy:
-                if game.clan.deputy.ID == self.ID:
-                    game.clan.deputy = None
-                    game.clan.deputy_predecessors += 1
-
-        elif self.status == "medicine cat":
+        elif new_status.is_medcat():
             if game.clan is not None:
                 game.clan.new_medicine_cat(self)
 
-        elif self.status == "elder":
-            if old_status == "leader":
+        elif new_status.is_warrior() or new_status.is_elder():
```

Comment:
Again very nitpicky, but why are we checking `new_status` instead of `self.status`? They might be equivalent in value, but since `self.status` is assigned the value of `new_status`, I'd much rather check that (like the old code did).
If you disagree, feel free to dispute this. I'm not hard sold on that.

Reply:
Honestly, I didn't notice this mistake. I _do_ think this is clearer for the reader, but it does introduce a small risk that somehow `self.status != new_status`.

Reply:
If there are no early returns in this method, you can assign `new_status` to `self.status` at the end instead, giving us both the certainty that the two don't diverge + the clearer intent.